packages/pluggableWidgets/video-player-native/src/VideoPlayer.tsx

```diff
@@ -36,27 +53,133 @@ export function VideoPlayer(props: VideoPlayerProps<VideoStyle>): ReactElement {
+                            paused={false}
+                            onLoad={data => {
+                                setStatus(StatusEnum.READY);
+                                setVideoAspectRatio(data.naturalSize.width / data.naturalSize.height);
+                                fullScreenPlayerRef.current?.seek(currentPlayTime);
+                            }}
+                            style={styles.fullScreenVideoStyle}
+                        />
+                    </TouchableWithoutFeedback>
+                    {showControls && (
+                        <TouchableOpacity
+                            style={styles.controlBtnContainerStyle}
+                            onPress={() => {
+                                fullScreenHandler(false);
+                            }}
+                        >
+                            <Icon name="fullscreen-exit" color="white" size={22} />
+                        </TouchableOpacity>
+                    )}
+                    {status === StatusEnum.LOADING && (
+                        <ActivityIndicator
+                            color={styles.indicator.color}
+                            size="large"
+                            style={styles.fullScreenActivityIndicatorStyle}
+                        />
+                    )}
+                    {status === StatusEnum.ERROR && (
+                        <Text style={styles.errorMessage}>The video failed to load :(</Text>
+                    )}
+                </Modal>
+            )}
+            <View style={styles.container}>
+                {status === StatusEnum.LOADING && <ActivityIndicator color={styles.indicator.color} size="large" />}
+                {status === StatusEnum.ERROR && <Text style={styles.errorMessage}>The video failed to load :(</Text>}
+                {!fullScreen && (
+                    <TouchableWithoutFeedback style={styles.container} onPress={() => handleShowControls()}>
+                        <Video
+                            {...videoProps}
+                            paused={!props.autoStart && !currentPlayTime}
+                            onLoad={data => {
+                                setStatus(StatusEnum.READY);
+                                setVideoAspectRatio(data.naturalSize.width / data.naturalSize.height);
+                                playerRef.current?.seek(currentPlayTime);
+                            }}
+                            ref={playerRef}
+                            style={status !== StatusEnum.READY ? { height: 0 } : styles.video}
+                            onProgress={({ currentTime }) => {
+                                setCurrentPlayTime(currentTime);
+                            }}
+                        />
+                    </TouchableWithoutFeedback>
+                )}
+                {isAndroid && showControls && (
+                    <TouchableOpacity
+                        onPress={() => {
```

Comment:
Nit: removing the { } reduces lines

Reply:
done

Reply:
It's still here :(

Reply:
I suggest using `useCallback`

Reply:
If `TouchableOpacity` uses `useMemo` this could be helpful